src/core/ext/filters/client_channel/resolver/dns/event_engine/event_engine_client_channel_resolver.cc

```diff
@@ -272,24 +272,27 @@ void EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
   {
     MutexLock lock(&on_resolved_mu_);
     orphaned_ = true;
-    // Event if cancellation fails here, OnResolvedLocked will return early, and
-    // the resolver will never see a completed request.
-    if (hostname_handle_.has_value()) {
-      event_engine_resolver_->CancelLookup(*hostname_handle_);
-    }
-    if (srv_handle_.has_value()) {
-      event_engine_resolver_->CancelLookup(*srv_handle_);
-    }
-    for (const auto& handle : balancer_hostname_handles_) {
-      event_engine_resolver_->CancelLookup(handle);
-    }
-    if (txt_handle_.has_value()) {
-      event_engine_resolver_->CancelLookup(*txt_handle_);
+    if (timeout_handle_.has_value()) {
+      resolver_->event_engine_->Cancel(*timeout_handle_);
+      timeout_handle_.reset();
     }
+    // Even if cancellation fails here, OnResolvedLocked will return early, and
```

Comment:
since the callback is now always invoked, does this callback still apply?

Reply:
Yes, I think it still apply. This part of the code hasn't been changed: https://github.com/grpc/grpc/blob/master/src/core/ext/filters/client_channel/resolver/dns/event_engine/event_engine_client_channel_resolver.cc#L475

And I think it's still the case that the resolver will not see a complete result after ee_client_channel_resolver is Orphaned.

Reply:
Got it , thanks (I was thinking about the DNSResolver cancellation rather than "client channel DNS resolver cancellation"